application.py

```diff
@@ -320,10 +320,24 @@ def dashboard():
 @app.route("/project/<project_name>", methods=["GET"])
 def view_project(project_name):
     headers = {"Authorization": f"{session['id_token']}"}
-    projects = requests.get(
-        f"{API_URL}/api/v1/projects/{project_name}",
+
+    project_list = requests.get(
```

Comment:
What about this?

```py
@app.route("/project/<project_name>", methods=["GET"])
def view_project(project_name):
    # Sanitize project name to prevent path traversal and injection attacks
    if not project_name or not re.match(r'^[a-zA-Z0-9_-]+$', project_name):
        flash("Invalid project name. Project names can only contain letters, numbers, hyphens and underscores.")
        return redirect(url_for("dashboard"))
    
    if len(project_name) > 100:
        flash("Project name is too long. Please try again.")
        return redirect(url_for("dashboard"))

    headers = {"Authorization": f"{session['id_token']}"}
    try:
        projects = requests.get(
            f"{API_URL}/api/v1/projects/{project_name}",
            headers=headers,
        ).json()
        
        # Check if project not found
        if projects.get("message") is None:
            flash("Project not found. Please try again.")
            return redirect(url_for("dashboard"))
            
        return render_template("view_project.html", project=projects)
        
    except requests.exceptions.RequestException as e:
        logger.error(f"Error fetching project: {str(e)}")
        flash("Error retrieving project. Please try again.")
        return redirect(url_for("dashboard"))
```

Reply:
I understand your point, but checking if the project exists accounts for a scenario in which the user doesn't click on a project from the dashboard and instead manually adds a project name to the end of the URL. Prior to this there was no checking as to whether a URL was valid before the request was made.

Reply:
Yeah I understand. Let's add validation for names when creating on the UI and on the API. I think simple regex validation is better than having another API request though.

Reply:
Added.